File: TravelBois2/WebApp/Models/Let.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using WebApp.Models;
using WebApp.Models.Enum;

namespace ServerApp.Models
{
    public class Let
    {
        [Key]
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public string DatumPolaska { get; set; }
        [Required]
        public string DatumDolaska { get; set; }
        [Required]
        public string VremePoletanja { get; set; }
        [Required]
        public string VremeSletanja { get; set; }
        [Required]
        public string MestoPolaska { get; set; }
        [Required]
        public string MestoDolaska { get; set; }
        public ICollection<Aerodrom> Presedanja { get; set; }
        [Required]
        public int RazdaljinaPutovanja { get; set; }
        [Required]
        public int TrajanjePutovanja { get; set; }
        [Required]
        public KlasaLeta KlasaLeta { get; set; }
        [Required]
        public TipLeta TipLeta { get; set; }
        //public ICollection<Ocena> Ocene { get; set; }
        [Required]
        public int CenaKarte { get; set; }
        [Required]
        public string Aviokompanija { get; set; }
    }
}

File: TravelBois2/WebApp/ClientApp/src/app/entities/objects/let.ts
import { Prtljag } from './prtljag';
import { Ocena } from '../misc/ocena';
import { TipPrtljaga } from 'src/app/_enums';
import { AvioKompanija } from './avio-kompanija';
import { Aerodrom } from './aerodrom';
import { Sediste } from './sediste';
import { Presedanje } from './presedanje';
import { element } from 'protractor';

enum TipLeta {
  oneWay = 0,
  multiCity = 1,
}

enum KlasaLeta {
  first = 0,
  economy = 1,
  bussiness = 2,
}
export class Let {
  id: number;
  datumPolaska: string;
  datumDolaska: string;
  vremePoletanja: string;
  vremeSletanja: string;
  mestoPolaska: string;
  mestoDolaska: string;
  presedanjaLeta: Array<Presedanje>;
  razdaljinaPutovanja: number;
  trajanjePutovanja: number; //u minutima
  ocene: Array<Ocena>;
  cenaKarte: number;
  tipLeta: TipLeta;
  klasaLeta: KlasaLeta;
  aviokompanija: string;

  constructor(aviokompanija,mestoPolaska, mestoDolaska, datumPolaska, vremePoletanja, datumDolaska, vremeSletanja, trajanjePutovanja, razdaljinaPutovanja,klasa, tipLeta, presedanja = [], cena){
  this.aviokompanija = aviokompanija;
  this.datumPolaska = datumPolaska;
  this.datumDolaska = datumDolaska;
  this.mestoPolaska = mestoPolaska;
  this.mestoDolaska = mestoDolaska;
  this.vremePoletanja = vremePoletanja;
  this.vremeSletanja = vremeSletanja;
    if (presedanja == null) {
      this.presedanjaLeta = new Array<Presedanje>();
    }
    else {
      this.presedanjaLeta = new Array<Presedanje>();
      presedanja.forEach(element => {
        this.presedanjaLeta.push(element);
      })
    }
  this.trajanjePutovanja = trajanjePutovanja;
  this.razdaljinaPutovanja = razdaljinaPutovanja;
    if (tipLeta == 0) {
      this.tipLeta = TipLeta.oneWay
    }
    else if (tipLeta == 1) {
      this.tipLeta = TipLeta.multiCity;
    }
    else if (tipLeta == 'oneWay') {
      this.tipLeta = TipLeta.oneWay;
    }
    else {
      this.tipLeta = TipLeta.multiCity
    }

    if (klasa == 1) {
      this.klasaLeta = KlasaLeta.economy;
    }
    else if (klasa == 0) {
      this.klasaLeta = KlasaLeta.first;
    }
    else if (klasa == 2) {
      this.klasaLeta = KlasaLeta.bussiness;
    }
    else if (klasa == 'first') {
      this.klasaLeta = KlasaLeta.first;
    }
    else if (klasa == 'economy') {
      this.klasaLeta = KlasaLeta.economy;
    }
    else {
      this.klasaLeta = KlasaLeta.bussiness;
    }

  this.cenaKarte = cena;
  this.ocene = new Array<Ocena>();
  }
}
